lib/core/services/auth-guard.service.ts

```diff
@@ -30,6 +30,31 @@ export class AuthGuard extends AuthGuardBase {
                 router: Router,
                 appConfigService: AppConfigService) {
         super(authenticationService, router, appConfigService);
+        window.addEventListener('storage', this.ticketChange.bind(this));
+    }
+
+    ticketChange(event: StorageEvent) {
+        if (event.key === 'ticket-ECM' && event.newValue !== event.oldValue) {
+            this.ticketChangeRedirect(event, 'ECM');
+        }
+
+        if (event.key === 'ticket-BPM' && event.newValue !== event.oldValue) {
+            this.ticketChangeRedirect(event, 'BPM');
+        }
+
+        if (event.key === 'access_token' && event.newValue !== event.oldValue) {
+            this.ticketChangeRedirect(event, 'ALL');
+        }
+    }
+
+    private ticketChangeRedirect(event: StorageEvent, provider: string) {
+        if (!event.newValue) {
+            this.redirectToUrl(provider, this.router.url);
+        } else {
+            window.location.reload();
+        }
+
+        window.removeEventListener('storage', this.ticketChange);
```

Comment:
I don't think it would work properly. You added the listener: `this.ticketChange.bind(this)`, which created a new function, so if you want to remove the `this.ticketChange` it won't remove anything, since it wasn't the thing which was added. If I understand it properly.

Reply:
I think you are right